### PR TITLE
Do not write settings loaded from environment values

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

- Do not write settings loaded from environment values
- Disable widgets in preferences that loaded from env variables

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [x] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
like it!

File: napari/_qt/dialogs/preferences_dialog.py [hunk 1]
```diff
@@ -230,9 +230,19 @@ def build_page_dialog(self, schema, values):
...
         builder = WidgetBuilder()
         form = builder.create_form(schema, self.ui_schema)
+
+        # Disable widgets that loaded settings from environment variables
+        section = schema["section"]
+        form_layout = form.widget.layout()
+        for row in range(form.widget.layout().rowCount()):
+            widget = form_layout.itemAt(row, form_layout.FieldRole).widget()
+            name = widget._name
+            widget.setDisabled(
```
Comment (ppwadhwa) [inline napari/_qt/dialogs/preferences_dialog.py:242]: This line isn't working for me.  If I do something like the following it works as expected: 

```
if SETTINGS._env_settings.get(section, {}).get(name, False) is not False:
                widget.setDisabled(True)
```

**Review by ppwadhwa — CHANGES_REQUESTED** [review]

File: napari/_qt/dialogs/preferences_dialog.py (hunk 1 above)
Comment (goanpeca) [reply napari/_qt/dialogs/preferences_dialog.py:242]: Will check @ppwadhwa, thanks

**Comment by ppwadhwa** [comment]
Besides that one line, it works great!

File: napari/_qt/dialogs/preferences_dialog.py (hunk 1 above)
Comment (goanpeca) [reply napari/_qt/dialogs/preferences_dialog.py:242]: Fixed!

**Review by ppwadhwa — APPROVED** [review]
LGTM!  :)

**Review by sofroniewn — APPROVED** [review]
LGTM! Thanks!!

**Pull request merged**